aten/src/ATen/native/cpu/PowKernel.cpp

```diff
@@ -34,126 +36,104 @@ void pow_tensor_tensor_kernel(TensorIteratorBase& iter) {
   }
 }
 
-void pow_tensor_scalar_kernel(TensorIteratorBase& iter, const Scalar& exp_scalar) {
-  if (isFloatingType(iter.dtype())) {
-    const auto exp = exp_scalar.to<double>();
-    // Floating types allow AVX2 vector optimizations for pow/sqrt/rsqrt:
-    AT_DISPATCH_FLOATING_TYPES(iter.dtype(), "pow", [&]() {
+// Floating and complex types (minus Half) allow for optimized kernels
+template <typename scalar_t, typename cast_scalar_t, typename exp_scalar_t>
```

Comment:
I've added the comment, but will modify it, if required. Thanks!

Reply:
> But Complex types' computation doesn't allow standard // & double-precision to be mixed.

@imaginary-person could you elaborate on this?

Reply:
Hello @anjali411, thanks for pointing this out.
`std::pow` can take either complex64 or complex128 inputs, but not both. 
I should revise this comment, in order to be precise.

Reply:
@imaginary-person that answers my question, thanks! Also just curious -- why is it useful to allow a `double` exponent for a floating point input. I guess this case wouldn't be triggered from a Python API call right?

Reply:
@anjali411, yes, I believe you're absolutely right, but it can be triggered with the C++ API, eg. in https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/test/pow_test.cpp.

Reply:
ok yeah that's what I thought! cool thanks and great work on the PR

Reply:
Thank you, @anjali411, but @heitorschueroff deserves the credit for doing the most impactful work in this PR.